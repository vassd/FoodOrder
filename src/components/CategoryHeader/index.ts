export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
