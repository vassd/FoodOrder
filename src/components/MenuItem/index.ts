export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
