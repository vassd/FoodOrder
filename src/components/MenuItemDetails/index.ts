export { MenuItemDetails } from './MenuItemDetails'
export type { Topping, Crust } from './MenuItemDetails'
