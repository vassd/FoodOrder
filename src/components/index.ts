export { MenuHeader } from './MenuHeader'
export { CategoryHeader } from './CategoryHeader'
export type { Category } from './CategoryHeader'
export { MenuItems } from './MenuItems'
export { MenuItem } from './MenuItem'
export type { MenuItemInterface } from './MenuItem'
