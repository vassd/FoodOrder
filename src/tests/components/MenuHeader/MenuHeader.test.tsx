import React from 'react'
import '@testing-library/jest-dom'
import { describe, expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import { MenuHeader } from 'components'

describe('MenuHeader test', () => {
  const testName: string = '3rd rest'

  test('should render correctly', () => {
    render(<MenuHeader name={testName} />)

    expect(screen.getByText(testName)).toBeInTheDocument()
  })

  test('should match snapshot', () => {
    const { container } = render(<MenuHeader name={testName} />)

    expect(container).toMatchSnapshot()
  })
})
