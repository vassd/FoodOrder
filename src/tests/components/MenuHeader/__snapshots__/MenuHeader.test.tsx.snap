// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuHeader test > should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container _menu-header_e0233c css-1vam7s3-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-md-10 _menu-header__title_e0233c css-1b2f4c9-MuiGrid-root"
    >
      <strong>
        3rd rest
      </strong>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-2 _menu-header__info_e0233c css-frxywp-MuiGrid-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="InfoOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
        />
      </svg>
    </div>
  </div>
</div>
`;
